holidays/countries/nepal.py

```diff
@@ -0,0 +1,366 @@
+        2021: (MAY, 14),
+        2022: (MAY, 3),
+        2023: (APR, 22),
+        2024: (APR, 11),
+        2025: (MAR, 31),
+    }
+
+
+class NepalHinduHolidays(_CustomHinduHolidays):
+    # Maghe Sankranti.
+    MAKAR_SANKRANTI_DATES = {
+        2022: (JAN, 15),
+        2023: (JAN, 15),
+        2024: (JAN, 15),
+        2026: (JAN, 15),
+    }
+
+    # Holi (Terai).
+    HOLI_DATES = {
+        2023: (MAR, 7),
+        2026: (MAR, 3),
+    }
+
+    # Ram Navami.
+    RAM_NAVAMI_DATES = {
+        2026: (MAR, 27),
+    }
+
+    # Nepal New Year.
+    VAISAKHI_DATES = {
+        2025: (APR, 14),
+    }
+
+    # Janai Punima.
+    RAKSHA_BANDHAN_DATES = {
+        2022: (AUG, 12),
+        2023: (AUG, 31),
+    }
+
+    # Shree Krishna Janmashtami.
+    JANMASHTAMI_DATES = {
+        2023: (SEP, 6),
+    }
+
+    # Maha Ashtami.
+    MAHA_ASHTAMI_DATES = {
+        2020: (OCT, 24),
+    }
+
+    # Maha Navami.
+    MAHA_NAVAMI_DATES = {
+        2020: (OCT, 25),
+    }
+
+    # Vijayadashami.
+    DUSSEHRA_DATES = {
+        2020: (OCT, 26),
+    }
+
+    # Govardhan Puja.
+    GOVARDHAN_PUJA_DATES = {
+        2020: (NOV, 16),
+        2022: (OCT, 26),
+    }
+
+    # Chhath Parwa.
+    CHHATH_PUJA_DATES = {
+        2025: (OCT, 27),
+    }
+
+
+class NP(Nepal):
+    pass
+
+
+class NPL(Nepal):
+    pass
+
+
+class NepalStaticHolidays:
+    """Nepal special holidays.
+
+    References:
+        * [Death of Krishna Prasad Bhattarai](https://myrepublica.nagariknetwork.com/news/27618/)
+        * [Death of Sushil Koirala](https://kathmandupost.com/miscellaneous/2016/02/09/cabinet-decision-koirala-to-be-honoured-with-state-funeral)
+        * [Crash of Yeti Airlines Flight 691](https://edition.cnn.com/2023/01/15/asia/nepal-yeti-airlines-crash-intl-hnk/index.html)
+        * [People's War Day Instituted](https://kathmandupost.com/national/2023/02/12/government-announces-public-holiday-on-monday-to-mark-people-s-war-day)
+        * [People's War Day Annulled](https://kathmandupost.com/national/2023/12/29/supreme-court-annuls-public-holiday-on-people-s-war-day)
+        * [Death of Subas Chandra Nembang](https://en.nepalkhabar.com/news/detail/6023/)
+    """
+
+    # Day of National Mourning.
+    name_day_of_national_mourning = "Day of National Mourning"
+
+    # Tihar Holiday.
+    name_tihar_holiday = "Tihar Holiday"
+
+    special_public_holidays = {
+        2011: (MAR, 6, name_day_of_national_mourning),
+        2016: (FEB, 10, name_day_of_national_mourning),
+        2021: (NOV, 7, name_tihar_holiday),
+        2022: (OCT, 28, name_tihar_holiday),
+        2023: (
+            (JAN, 16, name_day_of_national_mourning),
+            # People War's Day.
+            (FEB, 13, "People War's Day"),
+            (SEP, 14, name_day_of_national_mourning),
+            (NOV, 16, name_tihar_holiday),
+        ),
+        2024: (
+            (NOV, 1, name_tihar_holiday),
+            (NOV, 4, name_tihar_holiday),
+        ),
```

Comment:
```suggestion
        2024: (NOV, 4, name_tihar_holiday),
```
November 1 is usual Gai Tihar.